src/components/Particles/smoke/SmokeParticle.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useRef, useMemo } from "react";
+import { useLoader, useFrame } from "@react-three/fiber";
+import * as THREE from "three";
+
+export const SmokeParticle = ({ position, png, leftDrift, rightDrift, delay = 0 }) => {
+  const texture = useLoader(THREE.TextureLoader, png);
+  const pointsRef = useRef();
+  const initialized = useRef(false);
+
+  // Initialize after delay
+  useMemo(() => {
+    const timer = setTimeout(() => {
+      initialized.current = true;
+    }, delay);
+    return () => clearTimeout(timer);
+  }, [delay]);
+
+  const points = useMemo(() => {
+    const geom = new THREE.BufferGeometry();
+    geom.setAttribute(
+      "position",
+      new THREE.Float32BufferAttribute(position, 3)
+    );
+    return geom;
+  }, [position]);
+
+  useFrame(({clock}, delta) => {
+    if (!initialized.current) return;
+
+    const pointsCurrent = pointsRef.current;
+
+    if (leftDrift) {
+      //Set inclination
+      pointsCurrent.position.x -= 0.09 * delta * 144;
+      pointsCurrent.position.z += 0.1 * delta * 144;
+      
+      if(pointsCurrent.position.x < -1.8) {
+        pointsCurrent.position.y = 0;
+        pointsCurrent.position.z = 0;
+        pointsCurrent.position.x = 0;
+        pointsCurrent.material.opacity = 1.5;
+        pointsCurrent.material.size = 4;
+      }
+    
+      if(pointsCurrent.material.opacity > 0) {
+        pointsCurrent.material.opacity -= 0.01 * delta * 144;
+      }
+
+      if(pointsCurrent.material.size > 0) {
+        //Shrinking effect
+        pointsCurrent.material.size -= 0.1* delta * 144;
+      }
+    } else if (rightDrift) {
+      //Set inclination
+      pointsCurrent.position.x += 0.09 * delta * 144;
+      pointsCurrent.position.z += 0.1 * delta * 144;
+
+      if(pointsCurrent.position.x > 1.8) {
+        pointsCurrent.position.y = 0;
+        pointsCurrent.position.z = 0;
+        pointsCurrent.position.x = 0;
+        pointsCurrent.material.opacity = 1.5;
+        pointsCurrent.material.size = 4;
+      }
+    
+      if(pointsCurrent.material.opacity > 0) {
+        pointsCurrent.material.opacity -= 0.01 * delta * 144;
+      }
+
+      if(pointsCurrent.material.size > 0) {
+        //Shrinking effect
+        pointsCurrent.material.size -= 0.1* delta * 144;
+      }
+    } else {
+      pointsCurrent.position.y = 0;
+      pointsCurrent.position.z = 0;
+      pointsCurrent.position.x = 0;
+      pointsCurrent.material.opacity = 0;
+      pointsCurrent.material.size = 0;
+    }
```

Comment:
I recommend inverting this if statement (`if(!leftDrift && !rightDrift)`) also there are small differences between left & right drift (only `pointsCurrent.position.x` really) so you could have a small if statement for that, the rest can use the same code.

Reply:
It's a good observation, I prefered this solition because I didn't understand where i should use if(!leftDrift && !rightDrift)